hypervisor/arch/x86/guest/vioapic.c

```diff
@@ -393,14 +393,10 @@ vioapic_write(struct vioapic *vioapic, uint32_t addr, uint32_t data)
 			vioapic_send_intr(vioapic, pin);
 		}
 
-		/* remap for active: interrupt mask -> unmask
-		 * remap for deactive: interrupt mask & vector set to 0
+		/*
+		 * remap for any update to RTL
 		 */
-		data64 = vioapic->rtbl[pin].reg;
-		if ((((data64 & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMCLR)
-		  && ((last & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMSET))
-		  || (((data64 & IOAPIC_RTE_INTMASK) == IOAPIC_RTE_INTMSET)
-		  && ((vioapic->rtbl[pin].reg & IOAPIC_RTE_INTVEC) == 0))) {
+		if ( changed != 0U ) {
```

Comment:
@fyin1 can you have a review?

Reply:
From the log posted from Abdul, the RTE is changed from edge triggered to low level triggered. This is unusual usage. We need to dig more to find out why this operation happens with this device first.